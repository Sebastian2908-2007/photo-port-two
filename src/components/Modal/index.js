

function Modal({ onClose, currentPhoto }) {
    const {name, category, description, index} = currentPhoto;
    return (
     <div className="modalBackdrop">
         <div className="modalContainer">
             <h3 className="modalTitle">{name}</h3>
             <img src={require(`../../assets/large/${category}/${index}.jpg`)} alt="current-catagory"/>
             <p>{description}</p>
             <button onClick={onClose} type="button">
                 Close this Modal
             </button>
         </div>
     </div>
    );
};

export default Modal;