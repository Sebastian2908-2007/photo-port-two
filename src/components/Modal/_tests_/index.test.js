import { cleanup, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";
// mock my p hoto data
const currentPhoto = {
    name: 'Park bench',
    category: 'landscape',
    description: 'a beutiful bench alone in the middle of the woods...next to some stairs?',
    index: 1
};
const mockOnClose = jest.fn();
afterEach(cleanup);

describe('Model Component', () => {
    it('renders',() => {
        render(<Modal currentPhoto={currentPhoto} onClose={mockOnClose} />)
    });

    it('matches snapshot',() => {
        const { asFragment } = render(<Modal currentPhoto={currentPhoto} onClose={mockOnClose}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('test clickhandler',() => {
    it('calls onClose handler',() => {
        // get by text allows us to find an element by the text content *Render Modal*
        const { getByText } = render(<Modal onClose={mockOnClose} currentPhoto={currentPhoto} />);
        // simulate click event
        fireEvent.click(getByText('Close this Modal'));
        // Assert: Expected matcher
        expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
});
