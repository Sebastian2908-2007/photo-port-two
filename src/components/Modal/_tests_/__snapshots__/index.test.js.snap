// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current-catagory"
        src="1.jpg"
      />
      <p>
        a beutiful bench alone in the middle of the woods...next to some stairs?
      </p>
      <button
        type="button"
      >
        Close this Modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
